### ISSUE TITLE
Introduce module system

### ISSUE THREAD

**Issue by wabain** [DDMAL/diva.js#315]
Since working with thousand-line files isn't great, it would be good to break some code out into separate files. The utilities can all be split and up and the toolbar would be a good candidate as well. Since manually managing a namespace across a lot of files without explicit imports and exports is hard, I think the best way to approach this would be to introduce a module system.

**Comment by ahankinson** [comment]
While I agree in principle, I'm not so sure that it will be that easy in practice. Could you come up with a more concrete plan of action prior to implementing it?

**Comment by wabain** [comment]
Sure. I think the most convenient module format to use would be CommonJS style, which sidesteps issues with ES6 transpiling, forced strict mode, and so on. So the code would look like Node.js:

```js
var jquery = require('jquery');
var diva = require('./diva');

// ...

module.exports = whatever;
```

Here's the minimum I think is needed to match the current behaviour:

1. Configure a Webpack build to generate a standalone bundle which expects jQuery to be a global. This covers the use case where Diva is loaded through a `<script>` tag. Currently the bundled file is also useable as a CommonJS module, but I think in the new build bundlers would just look to the source files.
2. Add imports and exports to `diva.js`, `utils.js` and the plugins.\* Move the main export statement for the package from `utils.js` to `diva.js`.
2. Get the build process to use the Webpack config.
3. Integrate Webpack with Karma for the tests. There are two parts for this: getting Karma to load a Webpack bundle, and making the code coverage instrumentation aware of the mapping back to the source files.

Basically the only visible change from those steps would be that the source files could no longer be loaded in script tags. (Currently this works, although it leaks globals.)

Do you think this misses anything?

---

\* Longer-term, I think that the plugins should not have a dependency on the Diva core, so that they can be bundled separately. That means that they need to get access to `diva.Events` without an import. But that's pretty much a separate issue.
